### ISSUE TITLE
Editor perf metrics and basic optimization

### ISSUE THREAD

**Issue by doodlewind** [toeverything/blocksuite#1292]
Related:

* #1291 
* #1258 
* #1176
* #1244

**Comment by pengx17** [comment]
> Perf regression on heavy example
is related to https://github.com/toeverything/blocksuite/issues/1244. We may need to either find a way to fix the issue, or disable HMR by default.

**Comment by pengx17** [comment]
After Quill is replaced with Virgo, https://github.com/toeverything/blocksuite/pull/1413 is no longer needed and the heavy example loads much faster :tada:

**Comment by doodlewind** [comment]
Marked as completed after #1550 landed

**Comment by pengx17** [comment]
We can optimize the HMR plugin later and move it from this ticket
